### ISSUE TITLE
Heap memory leak

### ISSUE THREAD

**Issue by GanerCodes** [natanfudge/Not-Enough-Crashes#193]
After an hour or two this hashmap consumes all the heap memory
![image](https://github.com/user-attachments/assets/2351fd4b-8d21-478b-82a7-52d578505c15)

**Comment by natanfudge** [comment]
what minecraft versions are affected? what other mods are installed?

**Comment by GanerCodes** [comment]
im using 1.21, my modlist is:
```
2mal3s-recipes-v1.10.2b.jar
accessible-step-mc1.21-1.3.1.jar
AdditionalStructures-1.21-(v.5.0.0-fabric).jar
anyscale-1.0-1.21.jar
architectury-13.0.6-fabric.jar
Armorful-3.1.0+1.21.jar
ArmorPoser-fabric-1.21-6.0.9.jar
arts_and_crafts-fabric-1.21-1.1.8.jar
athena-fabric-1.21-4.0.0.jar
azurelib-fabric-1.21-2.3.15.jar
azurepaxels-fabric-1.21-1.0.3.jar
BadOptimizations-2.1.4-1.21.jar
bakers-tools-0.1.5.jar
balancedrecall-1.0.6.jar
balm-fabric-1.21-21.0.14.jar
basicstorage-1.0.1+1.21.jar
BeamBeGone-0.0.2+1.21.jar
better-babies-0.7.11.jar
BetterF3-11.0.1-Fabric-1.21.jar
better_mcdonalds_mod-fabric-4.2.0+1.21.jar
bettersafebed-fabric-1.21-16.jar
BetterThanBunnies-1.21-Fabric-1.4.0.jar
BiomesOPlenty-fabric-1.21-21.0.0.15.jar
bitsbobs-1.21-1.0.0.jar
blockus-2.9.1+1.21.jar
BOMD-1.10.1-1.21.jar
bottledair-1.21.1-2.4.jar
BowInfinityFix-1.21-fabric-3.1.0.jar
breakfree-1.3.0.jar
brrp-fabric-1.0.3-1.21.jar
c2me-fabric-mc1.21-0.2.0+alpha.11.107.jar
carryon-fabric-1.21-2.2.1.jar
cc-tweaked-1.21-fabric-1.112.0.jar
chisel-fabric-1.21-1.8.1.jar
Chunky-1.4.16.jar
cicada-lib-0.8.3+1.21-and-above.jar
cinderscapes-5.0.1.jar
cloth-config-15.0.128-fabric.jar
cluckifer-1.0.0+1.21.jar
Clumps-fabric-1.21-18.0.0.2.jar
cobweb-fabric-1.21-1.1.2.jar
collective-1.21.1-7.81.jar
coloredbricks-fabric-1.21-5.2.jar
CommandKeys-Fabric-1.21-2.1.0-beta.1.jar
componentviewer-fabric-1.1.2+1.21.1.jar
compressedblocks-fabric-1.21-1.7.0-1.21.0.jar
Configurable-mc1.21-v1.3.1.jar
cookyourfood--mc1.21--fabric--1.12.jar
craftable_enchanted_golden_apple-1.0.7.jar
craftable-potion-combinations-1.1.3+1.21.jar
cupboardfabric-1.21-2.7.jar
customizablecarts-0.2.3.jar
custom-portals-3.6.1-1.21.jar
cyberforged-1.0.8.jar
cyclepaintings-1.21.1-3.6.jar
datastacks-0.9.5.jar
deadly-deadly-dungeon-1.0.3.1.jar
Debugify-1.21+1.0.jar
dense-flowers-0.1.1+mc1.21.0.jar
discerning-furnace-1.1.0+MC1.21.jar
disenchanting_table-merged-1.21-3.0.2.jar
DistantHorizons-2.1.2-a-1.21-neo-fabric.jar
do_a_barrel_roll-fabric-3.6.1+1.21.jar
DoAFlip-fabric-1.21-4.0.0.jar
door_knocking-1.0.1+1.21-fabric.jar
dragonkind-evolved-1.5.1.jar
DramaticDoors-NeoQuiFab-1.21-3.2.4_1.jar
dungeons-and-taverns-v4.3.jar
dynamicassetgenerator-6.1.1-fabric.jar
EasyAnvils-v21.0.4-1.21-Fabric.jar
EasyMagic-v21.0.2-1.21-Fabric.jar
EffectTimerPlus-Fabric-1.21-2.2.0.jar
enchantingcommands-1.21.1-3.4.jar
enchlevel-langpatch-2.2.6.jar
endrem-fabric-1.21-6.0.0.jar
energizedpower-1.21-2.12.1-fabric.jar
enhancedblockentities-0.10.1+1.21.jar
entityculling-fabric-1.6.6-mc1.21.jar
excavatedvariants-4.3.0-fabric.jar
expanded-axe-enchanting-1.0.10.jar
expanded-bow-enchanting-1.1.1.jar
expanded-crossbow-enchanting-1.0.1.jar
expandedstorage-14.0.2+1.21-fabric.jar
expanded-trident-enchanting-1.0.10.jar
expanded-weapon-enchanting-1.1.0.jar
ExplorersCompass-1.21-2.2.5-fabric.jar
ExpOre-1.21-0.1.jar
extendedbonemeal-1.21.1-3.5.jar
extshape_blockus-fabric-2.2.0-mc1.21.jar
extshape-fabric-2.2.1-mc1.21.jar
fabric-api-0.102.0+1.21.jar
fabric-language-kotlin-1.12.0+kotlin.2.0.10.jar
fabric_wider_ender_chests-1.0.2-1.21.jar
factory_blocks+fabric+1.21-1.3.2.jar
FallingTree-1.21-1.21.0.7.jar
fasterrandom-5.1.0.jar
fastitems-1.0.0-1.21.jar
fastquit-3.0.0+1.20.6.jar
Fast Recipe-1.0.0+1.20.5-1.21.jar
FastScrolling-3.0.0.jar
ferritecore-7.0.0-fabric.jar
FeyTweaks-1.21-1.2.8.jar
FoodEffectTooltips-1.1.1-1.21.jar
forcecrawl-fabric-1.21-1.0.5.jar
ForgeConfigAPIPort-v21.0.8-1.21-Fabric.jar
fortunate-ancient-debris-3.jar
freecam-fabric-modrinth-1.3.0+mc1.21.jar
friendsandfoes-fabric-mc1.21-2.0.16.jar
FuelInfo-2.1+1.21.jar
fzzy_config-0.3.5+1.21.jar
gamemodeoverhaul-fabric-2.2.0+1.21.jar
ganer-expanded-armor.jar
ganermobfx-1.0.0.jar
geckolib-fabric-1.21-4.5.8.jar
ghastly-wail-1.2.0+MC1.21.jar
GlitchCore-fabric-1.21-2.0.0.2.jar
globalpacks-fabric-1.21-21.0.0.jar
goldenberry-fabric-1.21-1.0.0.jar
gravestones-1.0.9.jar
handcrafted-fabric-1.21-4.0.1.jar
healpgood-1.4.2.jar
hideexperimentalwarning-1.21.1-1.2.jar
HopoBetterMineshaft-[1.21]-1.2.7.jar
HopoBetterRuinedPortals-[1.21]-1.4.3.jar
horizontal_glass_panes-1.21-fabric-3.0.0.jar
human-reborn-1.0.8.jar
Iceball-1.4.1+mc1.21.jar
Iceberg-1.21-fabric-1.2.4.jar
iChunUtil-1.21-Fabric-1.0.0.jar
indium-1.0.34+mc1.21.jar
inventory-1.9.8.jar
InventoryCleaner-1.0.9-1.21.jar
inventorytotem-1.21.1-3.3.jar
iris-1.7.3+mc1.21.jar
ironfish-mcmeta-45_1.21.jar
ItemBorders-1.21-fabric-1.2.5.jar
Jade-1.21-Fabric-15.1.5.jar
jakes-build-tools-1.5.3.jar
justmobheads-1.21.1-8.3.jar
Kiwi-1.21-Fabric-15.1.0.jar
krypton-0.2.8.jar
lavender-0.1.11+1.21.jar
leatherbundle-2.1.0.jar
led-1.8.3.jar
letmedespawn-1.3.1.jar
lithium-fabric-mc1.21-0.12.7.jar
LOCAL-vein-mining.jar
logbegone-fabric-1.21-1.0.2.jar
mergedelytra-1.0.1.jar
mes-1.3.2-1.21.jar
midnightlib-fabric-1.5.7.jar
mine-enter-5.3.0-beta.1.jar
mingen-3.1.jar
mns-1.0.2-1.21.jar
modelfix-1.21-1.6-fabric.jar
moderate-loading-screen-1.4.2+1.21.jar
modernfix-fabric-5.19.0+mc1.21.jar
mod-loading-screen-1.0.4.jar
modmenu-11.0.1.jar
mods-command-mc1.21-1.1.7.jar
Mo-Glass-1.8.1-MC1.21.jar
monolib-fabric-1.21-1.3.0.jar
moreculling-1.21-0.26.0.jar
motionblur-1.0.5+mc1.21.jar
MouseTweaks-fabric-mc1.21-2.26.jar
mru-0.4.4+1.21.jar
mss-1.1.2-1.21.jar
musketmod-1.21-fabric-1.4.8.jar
mvs-4.1.3-1.21.jar
naturallychargedcreepers-1.21.1-3.5.jar
NaturesCompass-1.21-2.2.6-fabric.jar
naturespirit-1.6.0-1.21.jar
Neruina-2.1.1-fabric+1.21.jar
netheriteextras-0.4.0+mc1.21.jar
neutral-animals-1.2.6.jar
nicemod-1.4.2 - 1.21.jar
NMPR-fabric-1.21-3.0.0.jar
NoChatReports-FABRIC-1.21-v2.8.0.jar
nofeathertrample-1.21.1-1.3.jar
nohunger-fabric-1.21-1.3.1.jar
No Increasing Repair Cost-fabric-1.21-1.0.0.jar
noisium-fabric-2.2.1+mc1.21.jar
NoRealmsButton-1.1.0-fabric.jar
no-report-button-fabric-1.6.0.jar
NoShade-fabric-1.21-7.0.2.jar
no-teleport-cooldown-1.0.9.jar
nutritiousmilk-1.21.1-3.4.jar
OldCombatMod-0.1.3.jar
oldmclogo-1.1.3.jar
omnihopper-2.3.2+1.21.jar
OpenYourEye.jar
oritech-0.8.1.jar
OverlayTweaks 1.21-fabric-1.10.0.jar
owo-lib-0.12.11+1.21.jar
pack-png-painting-mcmeta-42-100_1.21.jar
panda-temple-V1.jar
particle_core-0.2.4+1.21.jar
PartySpore-1.0.1-1.21.0.jar
perspektive-1.4.0.jar
Philips-Ruins1.21-1.5[Fabric].jar
pneumonocore-1.1.6.jar
polydecorations-0.4.2+1.21.jar
polydex-1.2.3+1.21.jar
polyfactory-0.4.0-pre.5.1+1.21.jar
polymer-bundled-0.9.9+1.21.jar
polymorph-fabric-1.0.4+1.21.jar
preferred-gamerules-1.1.0+1.19.4.jar
Prism-1.21-fabric-1.0.8.jar
provihealth-2.1.0+1.21.jar
purpurpacks-pickaxe-effective-glass-2.0.jar
PuzzlesLib-v21.0.20-1.21-Fabric.jar
qraftys-bamboo-villages-2.2.jar
qraftys-bunkers-2.2.jar
qraftys-halloween-villages-2.2.jar
randombonemealflowers-1.21.1-4.6.jar
randommobsizes-fabric-1.21-2.2.0.jar
redbits-1.16.6.jar
reeses_sodium_options-1.7.3+mc1.21.jar
reinforced-barrels-2.5.1+1.21.jar
reinforced-shulker-boxes-3.1.0+1.21.jar
remember-my-txt-1.0.2+MC1.20.jar
removewardeneffect-0.0.2-1.20.5-6.jar
resourcefulconfig-fabric-1.21-3.0.3.jar
resourcefullib-fabric-1.21-3.0.9.jar
resourcelibrary-fabric-1.21-2.0.2.jar
resource_nether_ores-fabric-1.21-2.0.4.jar
respawneye-0.1.1+1.21_hotfix.jar
rotl-4.14-1.21-fabric.jar
RoughlyEnoughItems-16.0.744-fabric.jar
RoughlyEnoughProfessions-fabric-1.21-4.0.1.jar
rrls-5.0.6+mc1.21-fabric.jar
satin-2.0.0.jar
saturated-0.18.jar
ScalableLux-0.1.0+rc.1+fabric.43c9882-all.jar
seethroughlava-4.0-1.20.jar
sharpness6-1.21-2.jar
showdurability-1.1.1+1.21.0.jar
show-me-your-skin-1.11.2+1.21.jar
shulkerboxtooltip-fabric-5.0.4+1.21.jar
Simple-Block-Overlay 1.21-fabric-1.4.2.jar
simple-hud-enhanced+1.21-4.7.1.jar
simple-portal-gun-mc1.21-2.0.0.jar
sit-1.21-28.jar
sketch-1.21.1-1.2.1.jar
skinlayers3d-fabric-1.6.6-mc1.21.jar
skin_overrides-2.1.1+1.21.jar
skippy-pearls-1.21.jar
slab-back-1.2.3.jar
SlashLength-0.0.1+1.19.3+21.0.jar
snow-pig-4.0.2.jar
sodium-extra-0.5.7+mc1.21.jar
sodium-fabric-0.5.11+mc1.21.jar
Sounds-1.1.4+1.21.jar
spark-1.10.73-fabric.jar
sparsestructures-fabric-1.21-2.2.0.jar
staaaaaaaaaaaack-fabric-1.21-1.5-75.jar
StackDeobfuscatorFabric-1.4.3+08e71cc.jar
stal-alloys-1.2.3-for-1.21.jar
stonecutter-salvage-1.0.0.jar
stutterfix-mc1.21-0.2.3.jar
tectonic-fabric-1.21-2.4.1.jar
tep-fabric-1.21-1.0.0.jar
TerraBlender-fabric-1.21-4.0.0.1.jar
terrestria-7.0.0.jar
threatengl-fabric-1.3.3-release.1.jar
tinycoal-fabric-1.21-1.0.2.jar
tiptapshow-1.6.1+1.21.jar
tmb-1.1.0.jar
toms_mobs-2.0.0+1.21.jar
toms_peripherals_fabric-1.21-1.2.2.jar
too-many-shortcuts-0.0.4+mc.1.21.jar
torch-hit-fabric-1.21-7.0.0.jar
trashslot-fabric-1.21-21.0.2.jar
travelersbackpack-fabric-1.21-10.0.6.jar
traverse-8.0.0.jar
trowel-1.4.0.jar
twirl-0.3.3+1.21.jar
unbreakable-1.21-3.jar
up_and_away-1.0.0-beta.7+build.115.jar
useful-brush-0.2.0+mc1.21.0.jar
verdance-1.21.1-1.0.1.jar
VisualWorkbench-v21.0.3-1.21-Fabric.jar
vmp-fabric-mc1.21-0.2.0+beta.7.163-all.jar
wardentools-2.5.1+mc1.21.jar
wits-1.3.0+1.21-fabric.jar
wooltweaks-1.21.1-3.7.jar
worldweaver-21.0.12.jar
xtraarrows-3.4.2-fabric-mc1.21.jar
yeatthesheep-1.3.jar
YetAnotherConfigLib-3.5.0+1.21-fabric.jar
yggdrasil-structure-1.4.0.jar
Zoomify-2.14.0+1.21.jar
```

**Comment by natanfudge** [comment]
Fixed with 4.4.8
